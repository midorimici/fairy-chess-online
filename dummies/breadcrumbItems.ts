export const items = [
  {
    name: 'Home',
    path: '/',
  },
  {
    name: 'Docs',
    path: '/docs',
  },
  {
    name: 'Breadcrumb',
    path: '/docs/breadcrumb',
  },
];
