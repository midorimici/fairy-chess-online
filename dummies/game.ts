export const game = {
  id: '1234567890abcdef',
  name: 'Game name',
  description:
    'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
  initialBoardImage: {
    url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
    width: 256,
    height: 256,
  },
};
