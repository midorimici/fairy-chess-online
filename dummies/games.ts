export const games = [
  {
    id: '1234567890abcdef',
    name: 'First game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
  {
    id: '1234567890abcde0',
    name: 'Second game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
];
