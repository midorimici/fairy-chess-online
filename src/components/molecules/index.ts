export { BreadcrumbNav } from './BreadcrumbNav';
export { Container } from './Container';
export { GameList } from './GameList';
