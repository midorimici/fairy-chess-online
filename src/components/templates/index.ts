export { Game } from './Game';
export { Games } from './Games';
export { Home } from './Home';
export { NewRoom } from './NewRoom';
