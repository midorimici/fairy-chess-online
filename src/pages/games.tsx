import type { GetStaticProps, NextPage } from 'next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import type { Namespace } from 'react-i18next';
import { Games } from '~/components/templates';
import { fetchGql } from '~/lib/graphcms';

type Props = {
  games: Game[];
};

const GamesPage: NextPage<Props> = ({ games }) => <Games games={games} />;

export default GamesPage;

const namespaces: Namespace = ['common'];

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  const data = await fetchGql<GamesResponse>(`
    {
      games {
        id
        name
        description
        initialBoardImage {
          url
          width
          height
        }
      }
    }
  `);

  return {
    props: {
      games: data.games,
      ...(await serverSideTranslations(locale ?? 'en', namespaces)),
    },
  };
};
