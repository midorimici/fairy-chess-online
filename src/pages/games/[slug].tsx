import type { GetStaticPaths, GetStaticProps, NextPage } from 'next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import type { Namespace } from 'react-i18next';
import { i18n } from 'next-i18next.config';
import { Game } from '~/components/templates';
import { fetchGql } from '~/lib/hygraph';

const GamePage: NextPage<GameResponse> = ({ game }) => <Game game={game} />;

export default GamePage;

const namespaces: Namespace = ['common'];

export const getStaticProps: GetStaticProps<{ game: Game }, { slug: string }> = async ({
  params,
  locale,
}) => {
  if (params === undefined) {
    return {
      notFound: true,
    };
  }

  const data = await fetchGql<GameResponse>(`
    {
      game(where: {slug: "${params.slug}"}) {
        slug
        name
        description
        initialBoardImage {
          url
          width
          height
        }
      }
    }
  `);

  return {
    props: {
      game: data.game,
      ...(await serverSideTranslations(locale ?? 'en', namespaces)),
    },
  };
};

export const getStaticPaths: GetStaticPaths<{ slug: string }> = async () => {
  const data = await fetchGql<{ games: { slug: string }[] }>(`
    {
      games {
        slug
      }
    }
  `);

  const paths = data.games
    .map((game: { slug: string }) =>
      i18n.locales.map((locale: string) => ({ params: { slug: game.slug }, locale }))
    )
    .flat();

  return {
    paths,
    fallback: false,
  };
};
