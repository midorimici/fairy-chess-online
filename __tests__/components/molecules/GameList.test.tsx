import { render } from '@testing-library/react';
import { games } from 'dummies/games';
import { GameList } from '~/components/molecules';

it('renders GameList unchanged', () => {
  const { asFragment } = render(<GameList games={games} />);
  expect(asFragment()).toMatchSnapshot();
});
