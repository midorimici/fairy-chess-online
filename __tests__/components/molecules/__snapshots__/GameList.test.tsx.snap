// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GameList unchanged 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1clhiew"
  >
    <div
      class="chakra-linkbox css-1cnlftd"
    >
      <div
        class="css-2rlxtj"
      >
        <a
          class="chakra-linkbox__overlay css-15i5kzd"
          href="/games/first-game"
        >
          First game
        </a>
        <p
          class="chakra-text css-0"
        >
          Game desciption. This part may contains the description about the game or/and pieces used in the game.
        </p>
      </div>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjU2IiBoZWlnaHQ9IjI1NiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="chakra-linkbox css-1cnlftd"
    >
      <div
        class="css-2rlxtj"
      >
        <a
          class="chakra-linkbox__overlay css-15i5kzd"
          href="/games/second-game"
        >
          Second game
        </a>
        <p
          class="chakra-text css-0"
        >
          Game desciption. This part may contains the description about the game or/and pieces used in the game.
        </p>
      </div>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjU2IiBoZWlnaHQ9IjI1NiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
